src/panels/lovelace/cards/hui-conditional-card.ts

```diff
@@ -1,79 +1,22 @@
-import { createCardElement } from "../common/create-card-element";
-import { computeCardSize } from "../common/compute-card-size";
-import { HomeAssistant } from "../../../types";
 import { LovelaceCard } from "../types";
-import { LovelaceCardConfig } from "../../../data/lovelace";
-
-interface Condition {
-  entity: string;
-  state?: string;
-  state_not?: string;
-}
+import { LovelaceCardConfig, Condition } from "../../../data/lovelace";
+import { HuiConditional } from "../components/hui-conditional";
 
 interface Config extends LovelaceCardConfig {
   card: LovelaceCardConfig;
   conditions: Condition[];
 }
 
-class HuiConditionalCard extends HTMLElement implements LovelaceCard {
-  private _hass?: HomeAssistant;
-  private _config?: Config;
-  private _card?: LovelaceCard;
+class HuiConditionalCard extends HuiConditional implements LovelaceCard {
+  public isPanel?: boolean;
 
   public setConfig(config) {
-    if (
-      !config.card ||
-      !config.conditions ||
-      !Array.isArray(config.conditions) ||
-      !config.conditions.every((c) => c.entity && (c.state || c.state_not))
-    ) {
-      throw new Error("Error in card configuration.");
-    }
-
-    if (this._card && this._card.parentElement) {
-      this.removeChild(this._card);
-    }
-
-    this._config = config;
-    this._card = createCardElement(config.card);
-    if (this._hass) {
-      this.hass = this._hass;
-    }
-  }
-
-  set hass(hass: HomeAssistant) {
-    this._hass = hass;
-
-    if (!this._card) {
-      return;
-    }
-
-    const visible =
-      this._config &&
-      this._config.conditions.every((c) => {
-        if (!(c.entity in hass.states)) {
-          return false;
-        }
-        if (c.state) {
-          return hass.states[c.entity].state === c.state;
-        }
-        return hass.states[c.entity].state !== c.state_not;
-      });
-
-    if (visible) {
-      this._card.hass = hass;
-      if (!this._card.parentElement) {
-        this.appendChild(this._card);
-      }
-    } else if (this._card.parentElement) {
-      this.removeChild(this._card);
-    }
-    // This will hide the complete card so it won't get styled by parent
-    this.style.setProperty("display", visible ? "" : "none");
+    // todo: artifically using the above config but it's still important to distinguish between card config and the internal hui-conditional implementation config
+    super.setConditionalConfig(config as Config);
   }
 
   public getCardSize() {
-    return computeCardSize(this._card!);
+    return super.getCardSize();
```

Comment:
card size would only be relevant to the conditional card, not rows or elements, so not necessary to be in the base class